### DISCUSSION TITLE
[Feature Request] Expand Organization Migration functionality of GraphQL API to allow for a subset of repos to be included.

### DISCUSSION THREAD

**Discussion by dkoch84** [community/community#101929] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

It would not matter whether a user must supply a list of repos to include or exclude, but Organization migrations are much more useful and powerful than Repository migrations when using the GEI. 

Consider doing a very large, 1000s of repos, migration in stages. The Organization migration includes Teams and their permissions when moving repositories, whereas the Repository migration does not. Over many hundreds of repos and many teams, this can be a lot of work. The REST API's transfer endpoint makes transferring repos from one org to another very trivial, and thus, multiple full Org migrations is often the path of least resitance to moving 1000s of repos over a few stages. 

It would put less stress on the github APIs and infrastructure, as well as speed up operations, if we could say "do an Org migration, but leave out these repos _______".

**Comment by ghost** [comment]
Your feature request for extending the GitHub GraphQL API to support partial organization migrations, including or excluding a specific list of repositories, is indeed a valuable suggestion. It would certainly add flexibility and efficiency for those managing large numbers of repositories, especially when undergoing a phased migration.

Currently, as you've noted, the GitHub GraphQL API's organization migration functionality doesn't offer the granularity to include or exclude certain repositories. Instead, it's an all-or-nothing approach that can be less practical for very large organizations that want to migrate in stages.

I recommend submitting this feature request directly through GitHub's official feedback form, which is often reviewed by the product team for consideration in future updates. Here's how you can submit your request:

1. Go to the GitHub Feedback Discussions page.
2. Select the "API and Webhooks" category, as this is most relevant to your request.
3. Provide a clear and detailed explanation of your feature request, similar to the one you've posted here. Explain how it would benefit not just your workflow but also other GitHub users.
4. Highlight the potential for reducing stress on GitHub's APIs and infrastructure with this more efficient approach.
5. Submit the feedback and monitor the discussion for any updates from GitHub staff or community members.
In the meantime, a workaround might involve using the REST API to script the migration process by transferring repositories in batches. While not ideal, it could provide a stop-gap solution until such a feature is potentially implemented.

**Comment by ghostinhershell** [comment]
Hi @dkoch84, 

Thanks so much for reaching out in our community. I looked at our documentation and it looks like this may be possible using the API: 
> `repositories` - array of strings - Required
> A list of arrays indicating which repositories should be migrated.

I'm referencing this particular support doc: [Organization migrations](https://docs.github.com/en/rest/migrations/orgs?apiVersion=2022-11-28). Does this address what you're looking for here?

**Reply by dkoch84** [reply]
Hi, thanks for the reply. I believe the GitHub Enterprise Importer tooling utilizes the GraphQL API, whereas legacy tooling used the REST API. 

The REST API requires the downloading and then import of the migration tarball, whereas the GEI tooling queues the whole process on GitHub's servers, export and import. This is very much a hands off process and the recommended method to perform migrations, per Professional Services. 

See this tooling: https://github.com/github/gh-gei

**Reply by dkoch84** [reply]
I can further clarify. 

I have many means at my disposal to move many repos at once. The issue isn't that. The issue is that if you have an organization with 100 repos, and you need to move 75 of them, there's no method of moving just the 75 that takes less effort than migrating the whole Org and deleting 25 repos after. Then when you wanna migrate the remaining 25, it's still much less effort to do another Org migration and delete 75 repos. 

That's quite wasteful of compute resources. But it's very easy on humans. Takes longer than it should. 

To be able to run that "richer" full Org migration, but with a subset of repos, would be the path of least resistance in many scenarios. 

I should also say, this isn't some critical thing. Sure, I can leverage the API to copy the data that's otherwise not included in the Repo migrations. But I'm more technically experienced than a lot of folks trying to migrate. What to me is just another couple of scripts, is a whole extra ordeal to someone else. So to avoid that ordeal, more time and more data being moved is the tradeoff. It'd just be smart to make the Org migration endpoint a little flexible.
